pkg/controller/podprobemarker/pod_probe_marker_controller.go

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2022 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podprobemarker
+
+import (
+	"context"
+	"flag"
+	"reflect"
+	"strings"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util"
+	utilclient "github.com/openkruise/kruise/pkg/util/client"
+	"github.com/openkruise/kruise/pkg/util/controllerfinder"
+	"github.com/openkruise/kruise/pkg/util/ratelimiter"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	kubecontroller "k8s.io/kubernetes/pkg/controller"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+func init() {
+	flag.IntVar(&concurrentReconciles, "podprobemarker-workers", concurrentReconciles, "Max concurrent workers for PodProbeMarker controller.")
+}
+
+var (
+	concurrentReconciles = 3
+)
+
+const (
+	ReconPodProbeMarker = "PodProbeMarker"
+	ReconNodePodProbe   = "NodePodProbe"
+
+	// PodProbeMarkerFinalizer is used to remove podProbe from NodePodProbe.Spec
+	PodProbeMarkerFinalizer = "kruise.io/pod-probe-marker"
```

Comment:
plz comments about the finalizer is on PodProbeMarker, and consider rename to 
kruise.io/node-pod-probe-cleanup